azkaban-hadoop-security-plugin/src/main/java/azkaban/security/AbstractHadoopSecurityManager.java

```diff
@@ -169,35 +170,51 @@ private String verifySecureProperty(final Props props, final String s)
     return value;
   }
 
+  /**
+   * This method is used to get UGI for proxy user if userToProxyFQN is not possible to configure
+   * correctly.
+   * @param userToProxy
+   * @return
+   * @throws HadoopSecurityManagerException
+   */
+  private UserGroupInformation getProxiedUser(final String userToProxy)
+      throws HadoopSecurityManagerException {
+    return getProxiedUser(userToProxy, userToProxy);
+  }
+
   /**
    * Create a proxied user based on the explicit user name, taking other parameters necessary from
    * properties file.
+   *
+   * @param userToProxyFQN
```

Comment:
This might be a good place to explain the need and the meaning of the additional parameter. Otherwise, to someone who does not see the entire class hierarchy, only one of the two parameters is used here, and it's not clear why we need this change.
BTW, this is also a sign that structurally this is not optimal. We are changing this place *because* we know something else about somewhere else in the class hierarchy. So this class does not stand on its own. I'd have to think about a better way, and I guess we don't have time for it.